### DISCUSSION TITLE
Bypass the "view in ar" click

### DISCUSSION THREAD

**Discussion by roy-baker** [google/model-viewer#1920] [Q&A]
### Description

I have generated some QR codes to web pages that are model-viewer scenes. Is there a way to have the model-viewer code on the referenced web page automatically start the camera and ask for an anchor point (scan the floor)? This would bypass the necessity of having the users click the view in AR" button.

**Comment by elalish** [comment]
In fact no, this is not possible, and it's not something we can implement. For privacy and security reasons browsers do not allow websites to either start a webXR session or activate apps like SceneViewer without "user intent". So they intentionally require your users to click a button before going into AR. But you can customize that button to your heart's content using our `ar-button` slot.

**Comment by corecode1** [comment]
Actually, it's possible to launch ar by getting reference to ar button and calling `arButton.click()` or just `modelViewer.activateAR()`, it will work if you're doing it right after user gesture like custom button click, etc. 

I'd like to know, is it possible to launch ar without showing 3D view at all? The problem is, I have a very resource-intensive threejs application + model viewer on the same page. And I would like to use modelViewer for ar only to avoid loading models and textures into RAM twice, for my application and modelViewer (they're using different canvas elements and rendering contexts, so resources cannot be shared). Can I do it somehow?

**Reply by Aatmik-Coder** [reply]
can you please share a link of the code, how you achieved the above one.

**Comment by elalish** [comment]
Yes, take a look at my description in #1629 (you may need the forthcoming v1.3). As for `activateAR()`, just keep in mind that if it's not part of a user gesture, the browser will probably block it.
